std/string.d

```diff
@@ -5798,36 +5798,67 @@ unittest
  * function, or any of the conversion functions.
  *
  * Params:
- *     s = the string to check
+ *     s = the string or random access range to check
  *     bAllowSep = accept separator characters or not
  *
  * Returns:
  *     $(D bool)
  */
-bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
+bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S ||
+    (isRandomAccessRange!S &&
+    hasSlicing!S &&
+    isSomeChar!(ElementType!S) &&
+    !isInfinite!S))
 {
     import std.algorithm.comparison : among;
+    import std.ascii : isASCII;
 
-    if (s.empty)
+    // ASCII only case insensitive comparison with two ranges
+    static bool asciiCmp(S1)(S1 a, string b)
+    {
+        import std.algorithm.comparison : equal;
+        import std.algorithm.iteration : map;
+        import std.ascii : toLower;
+        import std.utf : byChar;
+        return a.map!toLower.equal(b.byChar.map!toLower);
+    }
+
+    // auto-decoding special case, we're only comparing characters
+    // in the ASCII range so there's no reason to decode
+    static if (isSomeString!S)
+    {
+        import std.utf : byCodeUnit;
+        auto codeUnits = s.byCodeUnit;
+    }
+    else
+    {
+        alias codeUnits = s;
+    }
+
+    if (codeUnits.empty)
         return false;
 
     // Check for NaN (Not a Number) and for Infinity
-    if (s.among!((a, b) => icmp(a, b) == 0)
+    if (codeUnits.among!((a, b) => asciiCmp(a.save, b))
             ("nan", "nani", "nan+nani", "inf", "-inf"))
         return true;
 
-    if (s.front == '-' || s.front == '+')
-        s.popFront;
+    auto frontResult = codeUnits.front;
+    if (frontResult == '-' || frontResult == '+')
+        codeUnits.popFront;
 
-    immutable iLen = s.length;
+    immutable iLen = codeUnits.length;
     bool bDecimalPoint, bExponent, bComplex, sawDigits;
 
     for (size_t i = 0; i < iLen; i++)
     {
-        immutable c = s[i];
+        immutable c = codeUnits[i];
+
+        if (!c.isASCII)
+            return false;
 
         // Digits are good, continue checking
-        // with the popFront character... ;)
+        // with the character
```

Comment:
"with the character" doesn't make sense to me, but "with the **popFront** character" doesn't make sense either. "with the **next** character"? Just drop the comment completely? It's pretty obvious what's going on.